added php 8.4 to wp 6.7 compat page

kittenkamala opened the pull request:
Issue #297 
Updated compatibility.md to include PHP 8.4 for WP v 6.7

javiercasares reviewed (changes requested):
MySQL and MariaDB have changed their versions.

kittenkamala commented:
Thanks Javier! Are you referring to versions being EOL'd ?

kittenkamala commented:
Okay so the original ticket is just referring to updating PHP compatibility for WP 6.7 and is not asking for updated info on EOL'd versions of MySQL or other software. I don't think those versions being EOL'd means that WP 6.7 is not compatible with them. According to [our own team announcement ](https://make.wordpress.org/hosting/2024/11/05/wordpress-6-7-server-compatibility/#:~:text=WordPress%206.7%20(released%20on%20November,(LTS)%2C%20and%209.1.)WP 6.7 is compatible with MySQL 8.0 and MariaDB 10.5 or higher even if some of those versions later on are EOL'd. So I am not sure why Javier's comment should prevent this from being merged. A new ticket can be created if the team wants EOL info to be updated here. I rebased this branch and added 2 new reviewers. Feel free to merge if nothing is wrong with this pull. TY

chaion07 reviewed (approved):
Looks good
